File: FlexTargetAndroid/app/src/main/java/com/flextarget/android/ui/drills/CQBDrillSummaryView.kt
package com.flextarget.android.ui.drills

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Check
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.flextarget.android.R
import com.flextarget.android.data.model.CQBShotResult
import com.flextarget.android.data.model.DrillRepeatSummary

@Composable
fun CQBDrillSummaryView(
    summaries: List<DrillRepeatSummary>,
    modifier: Modifier = Modifier
) {
    LazyColumn(modifier = modifier.fillMaxSize()) {
        items(summaries) { summary ->
            CQBDrillCard(summary = summary)
        }
    }
}

@Composable
private fun CQBDrillCard(summary: DrillRepeatSummary) {
    // Log the cqbPassed value
    android.util.Log.d("CQBDrillSummaryView", "Repeat #${summary.repeatIndex}: cqbPassed=${summary.cqbPassed}, cqbResults=${summary.cqbResults?.size ?: 0} results")
    
    Card(
        modifier = Modifier
            .fillMaxWidth()
            .padding(12.dp),
        elevation = CardDefaults.cardElevation(defaultElevation = 4.dp),
        colors = CardDefaults.cardColors(containerColor = Color(0xFF1a1a1a))
    ) {
        Column(modifier = Modifier.padding(16.dp)) {
            // Title
            Text(
                text = "Repeat #${summary.repeatIndex}",
                style = MaterialTheme.typography.headlineSmall,
                color = Color.White,
                fontWeight = FontWeight.Bold,
                fontSize = 18.sp,
                modifier = Modifier.padding(bottom = 16.dp)
            )

            // Metrics Section
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .background(Color(0xFF2a2a2a), shape = androidx.compose.foundation.shape.RoundedCornerShape(8.dp))
                    .padding(12.dp),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                // Total Shots Metric
                Column(
                    modifier = Modifier.weight(1f),
                    horizontalAlignment = Alignment.CenterHorizontally
                ) {
                    Text(
                        text = stringResource(R.string.cqb_total_shots),
                        color = Color.Gray,
                        fontSize = 12.sp,
                        style = MaterialTheme.typography.bodySmall
                    )
                    Text(
                        text = "${summary.numShots}",
                        color = Color.White,
                        fontSize = 24.sp,
                        fontWeight = FontWeight.Bold,
                        modifier = Modifier.padding(top = 4.dp)
                    )
                }

                // Duration Metric
                Column(
                    modifier = Modifier.weight(1f),
                    horizontalAlignment = Alignment.CenterHorizontally
                ) {
                    Text(
                        text = stringResource(R.string.cqb_duration),
                        color = Color.Gray,
                        fontSize = 12.sp,
                        style = MaterialTheme.typography.bodySmall
                    )
                    Text(
                        text = String.format("%.3f", summary.totalTime) + "s",
                        color = Color.White,
                        fontSize = 24.sp,
                        fontWeight = FontWeight.Bold,
                        modifier = Modifier.padding(top = 4.dp)
                    )
                }

                // Status Section
                Row(
                    modifier = Modifier.weight(1f),
                    horizontalArrangement = Arrangement.Center,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    val passed = summary.cqbPassed ?: false
                    if (passed) {
                        Icon(
                            imageVector = Icons.Default.Check,
                            contentDescription = "Passed",
                            tint = Color(0xFF4CAF50),
                            modifier = Modifier
                                .size(24.dp)
                                .padding(end = 4.dp)
                        )
                        Text(
                            text = stringResource(R.string.cqb_passed),
                            color = Color(0xFF4CAF50),
                            fontSize = 14.sp,
                            fontWeight = FontWeight.Bold
                        )
                    } else {
                        Text(
                            text = stringResource(R.string.cqb_failed),
                            color = Color.Red,
                            fontSize = 14.sp,
                            fontWeight = FontWeight.Bold
                        )
                    }
                }
            }

            // CQB Results
            summary.cqbResults?.let { results ->
                if (results.isNotEmpty()) {
                    Spacer(modifier = Modifier.height(16.dp))

                    Text(
                        text = stringResource(R.string.cqb_threat),
                        style = MaterialTheme.typography.titleMedium,
                        color = Color.White,
                        fontWeight = FontWeight.Bold
                    )
                    results.filter { it.isThreat }.forEach { result ->
                        TargetCardRow(result = result)
                    }

                    Spacer(modifier = Modifier.height(12.dp))

                    Text(
                        text = stringResource(R.string.cqb_non_threat),
                        style = MaterialTheme.typography.titleMedium,
                        color = Color.White,
                        fontWeight = FontWeight.Bold
                    )
                    results.filter { !it.isThreat }.forEach { result ->
                        TargetCardRow(result = result)
                    }
                }
            }
        }
    }
}

@Composable
private fun TargetCardRow(result: CQBShotResult) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .background(Color(0xFF2a2a2a), shape = androidx.compose.foundation.shape.RoundedCornerShape(6.dp))
            .padding(12.dp)
            .padding(vertical = 4.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.SpaceBetween
    ) {
        // Target Name
        Text(
            text = stringResource(
                id = when (result.targetName) {
                    "cqb_front" -> R.string.cqb_front
                    "cqb_swing" -> R.string.cqb_swing
                    "cqb_hostage" -> R.string.cqb_hostage
                    "disguised_enemy" -> R.string.disguised_enemy
                    else -> R.string.cqb // fallback
                }
            ),
            color = Color.White,
            fontSize = 14.sp
        )

        // Shots
        Text(
            text = stringResource(
                R.string.cqb_shots_format,
                result.actualValidShots,
                result.expectedShots
            ),
            color = Color.Gray,
            fontSize = 12.sp
        )

        // Status
        val statusText = when (result.cardStatus) {
            CQBShotResult.CardStatus.green -> "✓"
            CQBShotResult.CardStatus.red -> "✗"
        }
        Text(
            text = statusText,
            color = when (result.cardStatus) {
                CQBShotResult.CardStatus.green -> Color(0xFF4CAF50)
                CQBShotResult.CardStatus.red -> Color.Red
            },
            fontSize = 16.sp,
            fontWeight = FontWeight.Bold
        )
    }
}

File: FlexTargetiOS/flextarget/View/Drills/CQBDrillSummaryView.swift
import SwiftUI
import CoreData

struct CQBDrillSummaryView: View {
    let drillSetup: DrillSetup
    @State var summaries: [DrillRepeatSummary]
    
    @Environment(\.dismiss) private var dismiss
    @Environment(\.managedObjectContext) private var environmentContext
    
    private var viewContext: NSManagedObjectContext {
        if let coordinator = environmentContext.persistentStoreCoordinator,
           coordinator.persistentStores.isEmpty == false {
            return environmentContext
        }
        return PersistenceController.shared.container.viewContext
    }
    
    private var drillName: String {
        drillSetup.name ?? "Untitled Drill"
    }
    
    var body: some View {
        ZStack {
            Color.black.ignoresSafeArea()
            
            VStack(spacing: 0) {
                navigationBar
                
                if summaries.isEmpty {
                    emptyState
                } else {
                    ScrollView(.vertical, showsIndicators: false) {
                        VStack(spacing: 24) {
                            ForEach(summaries.indices, id: \.self) { index in
                                cqbDrillCard(for: summaries[index], index: index)
                            }
                        }
                        .padding(.vertical, 24)
                    }
                }
            }
        }
        .navigationBarHidden(true)
    }
    
    // MARK: - Navigation Bar
    
    private var navigationBar: some View {
        HStack {
            Button(action: { dismiss() }) {
                Image(systemName: "chevron.left")
                    .font(.system(size: 16, weight: .semibold))
                    .foregroundColor(.white)
            }
            Spacer()
            Text(drillName.capitalized)
                .font(.system(size: 16, weight: .bold))
                .foregroundColor(.white)
            Spacer()
            // Placeholder for alignment
            Image(systemName: "chevron.left")
                .font(.system(size: 16, weight: .semibold))
                .foregroundColor(.white)
                .opacity(0)
        }
        .padding(.horizontal, 20)
        .padding(.vertical, 16)
        .background(Color.black.opacity(0.6))
    }
    
    // MARK: - CQB Drill Card
    
    private func cqbDrillCard(for summary: DrillRepeatSummary, index: Int) -> some View {
        VStack(spacing: 16) {
            // Header: Repeat number and basic stats
            VStack(spacing: 12) {
                HStack {
                    Text(String(format: NSLocalizedString("repeat_number", comment: "Repeat number format"), summary.repeatIndex))
                        .font(.system(size: 16, weight: .bold))
                        .foregroundColor(.white)
                    Spacer()
                }
                
                // Stats row: Total shots and duration
                HStack(spacing: 16) {
                    VStack(spacing: 4) {
                        Text(NSLocalizedString("cqb_total_shots", comment: "Total shots label"))
                            .font(.system(size: 11, weight: .semibold))
                            .foregroundColor(.white.opacity(0.7))
                        Text("\(summary.numShots)")
                            .font(.system(size: 18, weight: .bold))
                            .foregroundColor(.white)
                    }
                    .frame(minWidth: 60)
                    
                    VStack(spacing: 4) {
                        Text(NSLocalizedString("cqb_duration", comment: "Duration label"))
                            .font(.system(size: 11, weight: .semibold))
                            .foregroundColor(.white.opacity(0.7))
                        Text(format(time: summary.totalTime))
                            .font(.system(size: 18, weight: .bold))
                            .foregroundColor(.white)
                    }
                    .frame(minWidth: 60)

                    Spacer()
                    
                    if let cqbPassed = summary.cqbPassed {
                        HStack(spacing: 6) {
                            Image(systemName: cqbPassed ? "checkmark.circle.fill" : "xmark.circle.fill")
                                .font(.system(size: 16, weight: .bold))
                            Text(cqbPassed ? NSLocalizedString("cqb_passed", comment: "Passed status") : NSLocalizedString("cqb_failed", comment: "Failed status"))
                                .font(.system(size: 12, weight: .bold))
                        }
                        .foregroundColor(cqbPassed ? .green : Color(red: 0.8705882352941177, green: 0.2196078431372549, blue: 0.13725490196078433))
                    }
                }
                .padding(.horizontal, 16)
                .padding(.vertical, 12)
                .background(Color.white.opacity(0.05))
                .cornerRadius(8)
            }
            .padding(16)
            .background(
                RoundedRectangle(cornerRadius: 12)
                    .fill(Color.white.opacity(0.03))
                    .overlay(
                        RoundedRectangle(cornerRadius: 12)
                            .stroke(Color.white.opacity(0.1), lineWidth: 1)
                    )
            )
            .padding(.horizontal, 20)
            
            // Card results: Target-by-target pass/fail
            if let cqbResults = summary.cqbResults, !cqbResults.isEmpty {
                VStack(spacing: 10) {
                    ForEach(cqbResults) { result in
                        targetCardRow(result)
                    }
                }
                .padding(.horizontal, 20)
            }
            
            // Watch replay button
            NavigationLink(destination: DrillReplayView(drillSetup: drillSetup, shots: summary.shots)) {
                HStack(spacing: 8) {
                    Image(systemName: "play.circle.fill")
                        .font(.system(size: 16, weight: .bold))
                    Text(NSLocalizedString("watch_replay_button", comment: "Watch replay button text"))
                        .font(.system(size: 14, weight: .bold))
                        .kerning(0.5)
                }
                .foregroundColor(Color(red: 0.8705882352941177, green: 0.2196078431372549, blue: 0.13725490196078433))
                .padding(.vertical, 10)
                .frame(maxWidth: .infinity)
                .background(
                    RoundedRectangle(cornerRadius: 12)
                        .fill(Color(red: 0.8705882352941177, green: 0.2196078431372549, blue: 0.13725490196078433).opacity(0.1))
                        .overlay(
                            RoundedRectangle(cornerRadius: 12)
                                .stroke(Color(red: 0.8705882352941177, green: 0.2196078431372549, blue: 0.13725490196078433).opacity(0.3), lineWidth: 1)
                        )
                )
            }
            .padding(.horizontal, 20)
        }
    }
    
    // MARK: - Target Card Row
    
    private func targetCardRow(_ result: CQBShotResult) -> some View {
        HStack(spacing: 12) {
            // Card indicator (green or red circle)
            Image(systemName: result.cardStatus == .green ? "checkmark.circle.fill" : "xmark.circle.fill")
                .font(.system(size: 18, weight: .bold))
                .foregroundColor(result.cardStatus == .green ? .green : Color(red: 0.8705882352941177, green: 0.2196078431372549, blue: 0.13725490196078433))
            
            // Target info
            VStack(alignment: .leading, spacing: 4) {
                HStack(spacing: 8) {
                    Text(result.targetName)
                        .font(.system(size: 14, weight: .semibold))
                        .foregroundColor(.white)
                    
                    if result.isThreat {
                        Text(NSLocalizedString("cqb_threat", comment: "Threat label"))
                            .font(.system(size: 10, weight: .bold))
                            .foregroundColor(.white)
                            .padding(.horizontal, 6)
                            .padding(.vertical, 2)
                            .background(Color.orange.opacity(0.3))
                            .cornerRadius(4)
                    } else {
                        Text(NSLocalizedString("cqb_non_threat", comment: "Non-threat label"))
                            .font(.system(size: 10, weight: .bold))
                            .foregroundColor(.white)
                            .padding(.horizontal, 6)
                            .padding(.vertical, 2)
                            .background(Color.blue.opacity(0.3))
                            .cornerRadius(4)
                    }
                }
                
                if let failureReason = result.failureReason {
                    Text(failureReason)
                        .font(.system(size: 12, weight: .regular))
                        .foregroundColor(.white.opacity(0.7))
                }
                
                // Shot count for threat targets
                if result.isThreat {
                    Text(String(format: NSLocalizedString("cqb_shots_format", comment: "Shots format"), result.actualValidShots, result.expectedShots))
                        .font(.system(size: 11, weight: .regular))
                        .foregroundColor(.white.opacity(0.6))
                }
            }
            
            Spacer()
        }
        .padding(12)
        .background(
            RoundedRectangle(cornerRadius: 8)
                .fill(result.cardStatus == .green ? Color.green.opacity(0.08) : Color(red: 0.8705882352941177, green: 0.2196078431372549, blue: 0.13725490196078433).opacity(0.08))
                .overlay(
                    RoundedRectangle(cornerRadius: 8)
                        .stroke(
                            result.cardStatus == .green ? Color.green.opacity(0.3) : Color(red: 0.8705882352941177, green: 0.2196078431372549, blue: 0.13725490196078433).opacity(0.3),
                            lineWidth: 1
                        )
                )
        )
    }
    
    // MARK: - Empty State
    
    private var emptyState: some View {
        VStack(spacing: 16) {
            Image(systemName: "exclamationmark.triangle.fill")
                .font(.system(size: 48))
                .foregroundColor(.white.opacity(0.5))
            Text(NSLocalizedString("no_results_title", comment: "No results title"))
                .font(.system(size: 18, weight: .bold))
                .foregroundColor(.white)
            Text(NSLocalizedString("no_results_message", comment: "No results message"))
                .font(.system(size: 14, weight: .regular))
                .foregroundColor(.white.opacity(0.6))
                .multilineTextAlignment(.center)
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
    }
    
    // MARK: - Helpers
    
    private func format(time: TimeInterval) -> String {
        let minutes = Int(time) / 60
        let seconds = Int(time) % 60
        let milliseconds = Int((time.truncatingRemainder(dividingBy: 1)) * 1000)
        
        if minutes > 0 {
            return String(format: "%d:%02d.%03d", minutes, seconds, milliseconds)
        } else {
            return String(format: "%.3fs", time)
        }
    }
}

#Preview {
    let mockSummary = DrillRepeatSummary(
        repeatIndex: 1,
        totalTime: 45.5,
        numShots: 8,
        firstShot: 0.5,
        fastest: 0.8,
        score: 0,
        shots: [],
        cqbResults: [
            CQBShotResult(targetName: "cqb_front", isThreat: true, expectedShots: 2, actualValidShots: 2, cardStatus: .green),
            CQBShotResult(targetName: "cqb_swing", isThreat: true, expectedShots: 2, actualValidShots: 1, cardStatus: .red, failureReason: "Missed 1 shot"),
        ],
        cqbPassed: false
    )
    
    let mockDrillSetup = DrillSetup(context: PersistenceController.preview.container.viewContext)
    mockDrillSetup.name = "CQB Drill"
    mockDrillSetup.mode = "cqb"
    
    return NavigationStack {
        CQBDrillSummaryView(drillSetup: mockDrillSetup, summaries: [mockSummary])
    }
}
